shiny/ui/_chat_normalize.py

```diff
@@ -231,11 +231,19 @@ def normalize_chunk(self, chunk: "dict[str, Any]") -> ChatMessage:
         return super().normalize_chunk(msg)
 
     def can_normalize(self, message: Any) -> bool:
-        if not isinstance(message, dict):
-            return False
-        if "message" not in message:
+        try:
+            from ollama import ChatResponse
+
+            # Ollama<0.4 used TypedDict (now it uses pydantic)
+            # https://github.com/ollama/ollama-python/pull/276
+            if isinstance(ChatResponse, dict):
+                return "message" in message and super().can_normalize(
+                    message["message"]
+                )
+            else:
+                return isinstance(message, ChatResponse)
```

Comment:
Is some of the context here that you have a message normalizer for Pydantic models?

Reply:
No, `DictNormalizer` will work for either case since they define `__getitem__()` on the pydantic model. I suppose that is a weird/subtle thing that requires extra context, and it'd be nice to take advantage of stronger pydantic typing, but I opted for the minimal change (especially if we're going to support older versions)